api/src/s2-api.ts

```diff
@@ -90,7 +90,62 @@ async function getPaper(s2Id: string, apiKey?: string): Promise<Paper | undefine
   );
   const response = await axios.get<S2ApiPaper>(`${apiOrigin}/paper/${s2Id}`, conf);
   const paper = toPaper(s2Id, response.data);
-  cache.set(s2Id, paper);
+  // only cache if the request is successful to avoid papers being permanent errors
+  if (response.status === 200) {
+    cache.set(s2Id, paper);
+  }
+
 
   return paper;
 }
+
+/**
+ * Fetches a single paper from the S2 API and converts it into a Paper,
+ * while also allowing for error handling using axios's promise interface.
+ *
+ * @param s2Id SHA of the paper to request
+ * @param apiKey optional key for higher rate limit
+ * @returns
+ */
+export function getPaperUncached(s2Id: string, apiKey?: string): AxiosPromise<Paper> {
+  if (!apiKey) {
+    console.warn(`
+      WARNING: The S2 Public API Key isn't set. If the backend makes more than 100 requests to
+      the public API in 5 minutes it'll be rate limited and might stop working.
+
+      See the README.md for more information about geting an API key.
+    `);
+  }
+
+  const conf: AxiosRequestConfig = {
+    headers: {
+      "user-agent":
+        "Semantic Reader API Client (https://scholarphi.semanticscholar.org)",
+    },
+    // cast to AxiosTransformer[] or else the base type is never[]
+    // TODO: See if a newer version of typescript is better at resolving empty array types
+    transformResponse: ([] as AxiosTransformer[])
```

Comment:
I don't remember having to do stuff like this back on Beaker, which uses TS 3.8, so I think if we took an updated version of TS there would be less of this going on.